In [ ]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import scipy.stats 
import scipy.stats as st
from scipy import stats

In [ ]:
data=pd.read_csv("https://raw.githubusercontent.com/labeconometria/MLxE/main/proyectos1er/dataset_2.csv")

#Introducción
En el siguiente proyecto se realizara el analisis de probabilidad y de estadistica inferencial dentro del dataset 2, el cual es una base de datos donde se busca que tipo de cliente se tiene, si uno existente o uno retirado.

#Tabla de contenido


1. Exploración Inicial y Limpieza de Datos
2. Verificación de la diferencia de porporciones en determinadas variables.
3. Verificación de distribucción normal en determinadas variables.
4. División del dataset para verificar la diferencia de medias en determinadas variables
5. Construcción de tablas de contingencia
6. Realización de la prueba $\chi^2$
7. Conclusión General


#1.Exploración Inicial y Limpieza de Datos

Para iniciar la exploración inicial obtenemos información acerca del dataset cargado. En este encontramos 10127 entradas y un total de 23 variables dentro de ella (columnas). A simple vista a traves de la información vemos que no se presentan datos faltantes pero se confirmara esto a continuación.  Así mismo vemos que de las 23 columnas de datos; 10 son de typo int, 6 de tipo object posiblemente dummies y 7 de tipo float.

In [ ]:
data

CLIENTNUM     Attrition_Flag  Customer_Age Gender  Dependent_count  \
0      768805383  Existing Customer            45      M                3   
1      818770008  Existing Customer            49      F                5   
2      713982108  Existing Customer            51      M                3   
3      769911858  Existing Customer            40      F                4   
4      709106358  Existing Customer            40      M                3   
...          ...                ...           ...    ...              ...   
10122  772366833  Existing Customer            50      M                2   
10123  710638233  Attrited Customer            41      M                2   
10124  716506083  Attrited Customer            44      F                1   
10125  717406983  Attrited Customer            30      M                2   
10126  714337233  Attrited Customer            43      F                2   

      Education_Level Marital_Status Income_Category Card_Category  \
0         High School        Married     $60K - $80K          Blue   
1            Graduate         Single  Less than $40K          Blue   
2            Graduate        Married    $80K - $120K          Blue   
3         High School        Unknown  Less than $40K          Blue   
4          Uneducated        Married     $60K - $80K          Blue   
...               ...            ...             ...           ...   
10122        Graduate         Single     $40K - $60K          Blue   
10123         Unknown       Divorced     $40K - $60K          Blue   
10124     High School        Married  Less than $40K          Blue   
10125        Graduate        Unknown     $40K - $60K          Blue   
10126        Graduate        Married  Less than $40K        Silver   

       Months_on_book  ...  Credit_Limit  Total_Revolving_Bal  \
0                  39  ...       12691.0                  777   
1                  44  ...        8256.0                  864   
2                  36  ...        3418.0                    0   
3                  34  ...        3313.0                 2517   
4                  21  ...        4716.0                    0   
...               ...  ...           ...                  ...   
10122              40  ...        4003.0                 1851   
10123              25  ...        4277.0                 2186   
10124              36  ...        5409.0                    0   
10125              36  ...        5281.0                    0   
10126              25  ...       10388.0                 1961   

       Avg_Open_To_Buy  Total_Amt_Chng_Q4_Q1  Total_Trans_Amt  Total_Trans_Ct  \
0              11914.0                 1.335             1144              42   
1               7392.0                 1.541             1291              33   
2               3418.0                 2.594             1887              20   
3                796.0                 1.405             1171              20   
4               4716.0                 2.175              816              28   
...                ...                   ...              ...             ...   
10122           2152.0                 0.703            15476             117   
10123           2091.0                 0.804             8764              69   
10124           5409.0                 0.819            10291              60   
10125           5281.0                 0.535             8395              62   
10126           8427.0                 0.703            10294              61   

       Total_Ct_Chng_Q4_Q1  Avg_Utilization_Ratio  \
0                    1.625                  0.061   
1                    3.714                  0.105   
2                    2.333                  0.000   
3                    2.333                  0.760   
4                    2.500                  0.000   
...                    ...                    ...   
10122                0.857                  0.462   
10123                0.683                  0.511   
10124     

In [ ]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10127 entries, 0 to 10126
Data columns (total 23 columns):
 #   Column                                                                                                                              Non-Null Count  Dtype  
---  ------                                                                                                                              --------------  -----  
 0   CLIENTNUM                                                                                                                           10127 non-null  int64  
 1   Attrition_Flag                                                                                                                      10127 non-null  object 
 2   Customer_Age                                                                                                                        10127 non-null  int64  
 3   Gender                                                                           

Obtenemos información de los nombres de columnas dentro del dataframe.

In [ ]:
data.columns

Index(['CLIENTNUM', 'Attrition_Flag', 'Customer_Age', 'Gender',
       'Dependent_count', 'Education_Level', 'Marital_Status',
       'Income_Category', 'Card_Category', 'Months_on_book',
       'Total_Relationship_Count', 'Months_Inactive_12_mon',
       'Contacts_Count_12_mon', 'Credit_Limit', 'Total_Revolving_Bal',
       'Avg_Open_To_Buy', 'Total_Amt_Chng_Q4_Q1', 'Total_Trans_Amt',
       'Total_Trans_Ct', 'Total_Ct_Chng_Q4_Q1', 'Avg_Utilization_Ratio',
       'Naive_Bayes_Classifier_Attrition_Flag_Card_Category_Contacts_Count_12_mon_Dependent_count_Education_Level_Months_Inactive_12_mon_1',
       'Naive_Bayes_Classifier_Attrition_Flag_Card_Category_Contacts_Count_12_mon_Dependent_count_Education_Level_Months_Inactive_12_mon_2'],
      dtype='object')

Para generar las estadisticas descriptivas del data frame usamos el metodo describe.

In [ ]:
data.describe()

CLIENTNUM  Customer_Age  Dependent_count  Months_on_book  \
count  1.012700e+04  10127.000000     10127.000000    10127.000000   
mean   7.391776e+08     46.325960         2.346203       35.928409   
std    3.690378e+07      8.016814         1.298908        7.986416   
min    7.080821e+08     26.000000         0.000000       13.000000   
25%    7.130368e+08     41.000000         1.000000       31.000000   
50%    7.179264e+08     46.000000         2.000000       36.000000   
75%    7.731435e+08     52.000000         3.000000       40.000000   
max    8.283431e+08     73.000000         5.000000       56.000000   

       Total_Relationship_Count  Months_Inactive_12_mon  \
count              10127.000000            10127.000000   
mean                   3.812580                2.341167   
std                    1.554408                1.010622   
min                    1.000000                0.000000   
25%                    3.000000                2.000000   
50%                    4.000000                2.000000   
75%                    5.000000                3.000000   
max                    6.000000                6.000000   

       Contacts_Count_12_mon  Credit_Limit  Total_Revolving_Bal  \
count           10127.000000  10127.000000         10127.000000   
mean                2.455317   8631.953698          1162.814061   
std                 1.106225   9088.776650           814.987335   
min                 0.000000   1438.300000             0.000000   
25%                 2.000000   2555.000000           359.000000   
50%                 2.000000   4549.000000          1276.000000   
75%                 3.000000  11067.500000          1784.000000   
max                 6.000000  34516.000000          2517.000000   

       Avg_Open_To_Buy  Total_Amt_Chng_Q4_Q1  Total_Trans_Amt  Total_Trans_Ct  \
count     10127.000000          10127.000000     10127.000000    10127.000000   
mean       7469.139637              0.759941      4404.086304       64.858695   
std        9090.685324              0.219207      3397.129254       23.472570   
min           3.000000              0.000000       510.000000       10.000000   
25%        1324.500000              0.631000      2155.500000       45.000000   
50%        3474.000000              0.736000      3899.000000       67.000000   
75%        9859.000000              0.859000      4741.000000       81.000000   
max       34516.000000              3.397000     18484.000000      139.000000   

       Total_Ct_Chng_Q4_Q1  Avg_Utilization_Ratio  \
count         10127.000000           10127.000000   
mean              0.712222               0.274894   
std               0.238086               0.275691   
min               0.000000               0.000000   
25%               0.582000               0.023000   
50%               0.702000               0.176000   
75%               0.818000               0.503000   
max               3.714000               0.999000   

       Naive_Bayes_Classifier_Attrition_Flag_Card_Category_Contacts_Count_12_mon_Dependent_count_Education_Level_Months_Inactive_12_mon_1  \
count                                       10127.000000                                                                                    
mean                                            0.159997                                                                                    
std                                             0.365301                                                                                    
min                                             0.000008                                                                                    
25%                                             0.000099                                                                                    
50%                                             0.000181                                                                                    
75%                                             

A partir del dataframe encontramos que la edad promedio de los clientes es de 46 años, para el tipo de cliente dependiente la media es de 2, para el Limite de Credito la media es de 8631 . Siendo estas variables las que consideramos relevante. Adicionalmente observamos que no se obtiene resultado en variables como "Attrition Flag", , el nivel de educación o si esta casado debido a que estas variables como lo demostro la información del dataframe son dummies.

Realizamos un head para explorar acerca de las 5 primeras columnas del dataset y un tail para observar las 5 ultimas columnas.

In [ ]:
data.head()

CLIENTNUM     Attrition_Flag  Customer_Age Gender  Dependent_count  \
0  768805383  Existing Customer            45      M                3   
1  818770008  Existing Customer            49      F                5   
2  713982108  Existing Customer            51      M                3   
3  769911858  Existing Customer            40      F                4   
4  709106358  Existing Customer            40      M                3   

  Education_Level Marital_Status Income_Category Card_Category  \
0     High School        Married     $60K - $80K          Blue   
1        Graduate         Single  Less than $40K          Blue   
2        Graduate        Married    $80K - $120K          Blue   
3     High School        Unknown  Less than $40K          Blue   
4      Uneducated        Married     $60K - $80K          Blue   

   Months_on_book  ...  Credit_Limit  Total_Revolving_Bal  Avg_Open_To_Buy  \
0              39  ...       12691.0                  777          11914.0   
1              44  ...        8256.0                  864           7392.0   
2              36  ...        3418.0                    0           3418.0   
3              34  ...        3313.0                 2517            796.0   
4              21  ...        4716.0                    0           4716.0   

   Total_Amt_Chng_Q4_Q1  Total_Trans_Amt  Total_Trans_Ct  Total_Ct_Chng_Q4_Q1  \
0                 1.335             1144              42                1.625   
1                 1.541             1291              33                3.714   
2                 2.594             1887              20                2.333   
3                 1.405             1171              20                2.333   
4                 2.175              816              28                2.500   

   Avg_Utilization_Ratio  \
0                  0.061   
1                  0.105   
2                  0.000   
3                  0.760   
4                  0.000   

   Naive_Bayes_Classifier_Attrition_Flag_Card_Category_Contacts_Count_12_mon_Dependent_count_Education_Level_Months_Inactive_12_mon_1  \
0                                           0.000093                                                                                    
1                                           0.000057                                                                                    
2                                           0.000021                                                                                    
3                                           0.000134                                                                                    
4                                           0.000022                                                                                    

   Naive_Bayes_Classifier_Attrition_Flag_Card_Category_Contacts_Count_12_mon_Dependent_count_Education_Level_Months_Inactive_12_mon_2  
0                                            0.99991                                                                                   
1                                            0.99994                                                                                   
2                                            0.99998                                                                                   
3                                            0.99987                                                                                   
4                                            0.99998                                                                                   

[5 rows x 23 columns]

In [ ]:
data.tail()

CLIENTNUM     Attrition_Flag  Customer_Age Gender  Dependent_count  \
10122  772366833  Existing Customer            50      M                2   
10123  710638233  Attrited Customer            41      M                2   
10124  716506083  Attrited Customer            44      F                1   
10125  717406983  Attrited Customer            30      M                2   
10126  714337233  Attrited Customer            43      F                2   

      Education_Level Marital_Status Income_Category Card_Category  \
10122        Graduate         Single     $40K - $60K          Blue   
10123         Unknown       Divorced     $40K - $60K          Blue   
10124     High School        Married  Less than $40K          Blue   
10125        Graduate        Unknown     $40K - $60K          Blue   
10126        Graduate        Married  Less than $40K        Silver   

       Months_on_book  ...  Credit_Limit  Total_Revolving_Bal  \
10122              40  ...        4003.0                 1851   
10123              25  ...        4277.0                 2186   
10124              36  ...        5409.0                    0   
10125              36  ...        5281.0                    0   
10126              25  ...       10388.0                 1961   

       Avg_Open_To_Buy  Total_Amt_Chng_Q4_Q1  Total_Trans_Amt  Total_Trans_Ct  \
10122           2152.0                 0.703            15476             117   
10123           2091.0                 0.804             8764              69   
10124           5409.0                 0.819            10291              60   
10125           5281.0                 0.535             8395              62   
10126           8427.0                 0.703            10294              61   

       Total_Ct_Chng_Q4_Q1  Avg_Utilization_Ratio  \
10122                0.857                  0.462   
10123                0.683                  0.511   
10124                0.818                  0.000   
10125                0.722                  0.000   
10126                0.649                  0.189   

       Naive_Bayes_Classifier_Attrition_Flag_Card_Category_Contacts_Count_12_mon_Dependent_count_Education_Level_Months_Inactive_12_mon_1  \
10122                                           0.000191                                                                                    
10123                                           0.995270                                                                                    
10124                                           0.997880                                                                                    
10125                                           0.996710                                                                                    
10126                                           0.996620                                                                                    

       Naive_Bayes_Classifier_Attrition_Flag_Card_Category_Contacts_Count_12_mon_Dependent_count_Education_Level_Months_Inactive_12_mon_2  
10122                                           0.999810                                                                                   
10123                                           0.004729                                                                                   
10124                                           0.002118                                                                                   
10125                                           0.003294                                                                                   
10126                                           0.003377                                                                                   

[5 rows x 23 columns]

Para iniciar la limpieza de datos del dataset iniciamos comprobando si este presenta missing values en alguna variable.
Como resultados observamos que el isna no encontro ningun dato faltante dentro del dataset.

In [ ]:
data.isna().sum()

CLIENTNUM                                                                                                                             0
Attrition_Flag                                                                                                                        0
Customer_Age                                                                                                                          0
Gender                                                                                                                                0
Dependent_count                                                                                                                       0
Education_Level                                                                                                                       0
Marital_Status                                                                                                                        0
Income_Category                                 

Verificamos así mismo que el dataset no presente datos duplicados y buscamos duplicados implicitos

In [ ]:
data.duplicated().sum()

0

In [ ]:
for i in data.columns:
  print(i)
  print(data[i].nunique()) 
  print(data[i].unique())

CLIENTNUM
10127
[768805383 818770008 713982108 ... 716506083 717406983 714337233]
Attrition_Flag
2
['Existing Customer' 'Attrited Customer']
Customer_Age
45
[45 49 51 40 44 32 37 48 42 65 56 35 57 41 61 47 62 54 59 63 53 58 55 66
 50 38 46 52 39 43 64 68 67 60 73 70 36 34 33 26 31 29 30 28 27]
Gender
2
['M' 'F']
Dependent_count
6
[3 5 4 2 0 1]
Education_Level
7
['High School' 'Graduate' 'Uneducated' 'Unknown' 'College' 'Post-Graduate'
 'Doctorate']
Marital_Status
4
['Married' 'Single' 'Unknown' 'Divorced']
Income_Category
6
['$60K - $80K' 'Less than $40K' '$80K - $120K' '$40K - $60K' '$120K +'
 'Unknown']
Card_Category
4
['Blue' 'Gold' 'Silver' 'Platinum']
Months_on_book
44
[39 44 36 34 21 46 27 31 54 30 48 37 56 42 49 33 28 38 41 43 45 52 40 50
 35 47 32 20 29 25 53 24 55 23 22 26 13 51 19 15 17 18 16 14]
Total_Relationship_Count
6
[5 6 4 3 2 1]
Months_Inactive_12_mon
7
[1 4 2 3 6 0 5]
Contacts_Count_12_mon
7
[3 2 0 1 4 5 6]
Credit_Limit
6205
[12691.  8256.  3418. ...  5409.  5281. 10

Observamos que el dataset no presenta datos duplicados, todos sus datos son unicos y podriamos afirmar tambien que no presenta datos faltantes. La limpieza esta completa para comenzar a trabajar con el dataset.

#3.Verificación de la diferencia de proporciones para las siguientes variables es estadísticamente significativa:
*   `Attrition Flag`
*   `Gender`





Importamos el estadistico de prueba ztest (Z) para poder realizar la comprobación de la significancia de las propociones

In [ ]:
from statsmodels.stats.weightstats import ztest as ztest

##Attrition Flag

Calculamos la proporción de cada dato dentro de la variable Attrition flag. Observamos que para Existing Customer la proporción es de 84% aproximadamente y para Attrited Customer la proporción es de 16%

In [ ]:
data['Attrition_Flag'].value_counts(normalize=True)

Existing Customer    0.83934
Attrited Customer    0.16066
Name: Attrition_Flag, dtype: float64

Capturamos en un nuevo dataframe la columna Attrition Flag considerando el dato "Existing Customer" como 1 y "Attrited Customer" como 0

In [ ]:
dataExisting=data["Attrition_Flag"]=="Existing Customer"
dataExisting=dataExisting.replace({True:1,False:0})
dataExisting

0        1
1        1
2        1
3        1
4        1
        ..
10122    1
10123    0
10124    0
10125    0
10126    0
Name: Attrition_Flag, Length: 10127, dtype: int64

Relizamosnuevamente el proceso anterior en un nuvo dataframe pero ahora considerando  "Attrited Customer" como 1 y "Existing Customer" como 0 

In [ ]:
dataAttrited=data["Attrition_Flag"]=="Attrited Customer"
dataAttrited=dataAttrited.replace({True:1,False:0})
dataAttrited

0        0
1        0
2        0
3        0
4        0
        ..
10122    0
10123    1
10124    1
10125    1
10126    1
Name: Attrition_Flag, Length: 10127, dtype: int64

Pruebas de hipotesis para verificar si la diferencia de proporciones para la variable Attrittion Flag es estadísticamente significativa:
*   Ho:La diferencia de las proporciones de las variables dentro de Attrition flag no es significativa 
*   Ha:La  diferencia de las proporciones de las variables dentro de de Attrition flag es significativa 

α=0.05

In [ ]:
ztest(dataExisting,dataAttrited, value=0, alternative='two-sided')

(131.5064542625507, 0.0)

Como resultado de la prueba  Z se compara el p valor y se puede decir que el p-valor es menor a 0.05 por lo tanto hay suficiente evidencia estadistica para considerar que la diferencia de proporciones de las variables dentro de Attrition Flag ( Existing Customer y Attried Customer) es significativa , concluyendo así que se rechaza la hipotesis nula.

##Gender

Calculamos la proporción de cada dato dentro de la variable Gender.Observamos que para F ( Female) la proporción es de 53% aproximadamente y para M (Male) la proporción es de 47%.

In [ ]:
data["Gender"].value_counts(normalize=True)

F    0.529081
M    0.470919
Name: Gender, dtype: float64

Capturamos en una un nuevo dataframe la columna Gender considerando M como 1 y F como 0

In [ ]:
dataM=data["Gender"]=="M"
dataM=dataM.replace({True:1,False:0})
dataM

0        1
1        0
2        1
3        0
4        1
        ..
10122    1
10123    1
10124    0
10125    1
10126    0
Name: Gender, Length: 10127, dtype: int64

Relizamos el proceso nuevamente en un nuevo dtaframe pero ahora considerando a F como 1 y a M como 0

In [ ]:
dataF=data["Gender"]=="F"
dataF=dataF.replace({True:1,False:0})
dataF

0        0
1        1
2        0
3        1
4        0
        ..
10122    0
10123    0
10124    1
10125    0
10126    1
Name: Gender, Length: 10127, dtype: int64

Pruebas de hipotesis para verificar si la diferencia de proporciones para Gender es
estadísticamente significativa:
*   Ho:La diferencia de las proporciones de las variables dentro de Gender no es significativa.
*   Ha:La  diferencia de las proporciones de las variables dentro de Gender es significativa.

α=0.05


In [ ]:
ztest(dataF,dataM, value=0, alternative='two-sided')

(8.290948979839161, 1.1234847132165844e-16)

Como resultado de la prueba Z se compara el p valor y se puede decir que el p-valor es menor a 0.05 por lo tanto si hay suficiente evidencia estadistica para considerar que la diferencia de proporciones de las variables dentro de Gender ( M y F) es significativa , concluyendo así que se rechaza la hipotesis nula.

#4.Verifique si las siguientes variables siguen una distribución normal:
*   `Customer Age`
*   `Dependent Count`







Para verificar si las variables presentan una distribución normal se necesita seleccionar las columnas de dichas variables. Esto se realiza a traves del metodo loc. Estas selecciones se nombran en un nuevo dataframe.

In [ ]:
data_customer=data.loc[:,"Customer_Age"]
data_dependent=data.loc[:,"Dependent_count"]

In [ ]:
data_customer

0        45
1        49
2        51
3        40
4        40
         ..
10122    50
10123    41
10124    44
10125    30
10126    43
Name: Customer_Age, Length: 10127, dtype: int64

In [ ]:
data_dependent

0        3
1        5
2        3
3        4
4        3
        ..
10122    2
10123    2
10124    1
10125    2
10126    2
Name: Dependent_count, Length: 10127, dtype: int64

A partir de los data frames seleccionados con dichas variables se realiza la prueba de Shapiro para determinar si las variables siguen una distribucción normal.

Prueba de Hipotesis de Distribucción Normal.
*   Ho:La distribución es normal
*   Ha:La distribución no es normal

α=0.05

In [ ]:
st.shapiro(data_customer)

/usr/local/lib/python3.7/dist-packages/scipy/stats/morestats.py:1760: UserWarning: p-value may not be accurate for N > 5000.
  warnings.warn("p-value may not be accurate for N > 5000.")


ShapiroResult(statistic=0.9960706233978271, pvalue=6.600162251926681e-16)

In [ ]:
st.shapiro(data_dependent)

ShapiroResult(statistic=0.936938464641571, pvalue=0.0)

Al realizar la prueba en ambas variables vemos que el p valor es menor (es muy pequeño o cercano a cero) a 0,05 en ambos casos, por lo que rechazamos la hipotesis nula y suponemos que estas variables no siguen una distribucción normal. 


#5.Divida el dataset en dos grupos a partir de la variable `Attrition Flag` . Verifique si hay diferencias en la media o en la mediana significativas en cuanto a `Customer Age` y en cuanto a `Dependent Count` para los dos grupos.

Se realiza la primera selección de las variables dentro del dataframe. Para esto creamos un nuevo data llamado exiscus que representa los Existing Customer & Customer Age. Así mismo está exisdepen que  representa los Existing Customer & Dependent Count.

In [ ]:
data_exiscus=data.loc[data["Attrition_Flag"]=="Existing Customer",["Customer_Age"]].reset_index(drop=True)
data_exisdepen=data.loc[data["Attrition_Flag"]=="Existing Customer",["Dependent_count"]].reset_index(drop=True)

A continuación se realiza la segunda selección de las variables dentro del dataframe. Para esto creamos un nuevo data llamado attriscus que representa los Attrited Customer & Customer Age. 
Así  está attridepen que  represneta los Attrited Customer & Dependent Count.

In [ ]:
data_attricus=data.loc[data["Attrition_Flag"]=="Attrited Customer",["Customer_Age"]].reset_index(drop=True)
data_attridepen=data.loc[data["Attrition_Flag"]=="Attrited Customer",["Dependent_count"]].reset_index(drop=True)

A partir de la división o selección de los data frame encontrariamos que se ubican de la siguiente manera 

In [ ]:
data_exiscus

Customer_Age
0               45
1               49
2               51
3               40
4               40
...            ...
8495            46
8496            57
8497            54
8498            56
8499            50

[8500 rows x 1 columns]

In [ ]:
data_attricus

Customer_Age
0               62
1               66
2               54
3               56
4               48
...            ...
1622            55
1623            41
1624            44
1625            30
1626            43

[1627 rows x 1 columns]

In [ ]:
data_exisdepen

Dependent_count
0                   3
1                   5
2                   3
3                   4
4                   3
...               ...
8495                5
8496                2
8497                1
8498                1
8499                2

[8500 rows x 1 columns]

In [ ]:
data_attridepen

Dependent_count
0                   0
1                   0
2                   1
3                   2
4                   2
...               ...
1622                3
1623                2
1624                1
1625                2
1626                2

[1627 rows x 1 columns]

Para realizar la prueba de hipotesis de diferencia de medias se tiene:
*   Ho: La diferencia de medias es igual 
*   Ha: La diferencia de medias es diferente



Corremos el codigo de diferencia de medias entre attricus( Attrited & CustomerAge) y exiscus (Existing Customer & CustomerAge).

In [ ]:
data_attricus.mean(),data_exiscus.mean()

(Customer_Age    46.659496
 dtype: float64, Customer_Age    46.262118
 dtype: float64)

Concluimos que las medias obtenidas a partir de la variable Attrition Flag en cuanto a Customer Age son diferentes , diciendo así que rechazamos la hipotesis nula en favor de la hipotesis alternativa.

Corremos ahora el codigo de diferencia de medias entre attridepen( Attrited&Dependent Count) y exisdepen (Existing Customer&Dependent Count).

In [ ]:
data_attridepen.mean(),data_exisdepen.mean()

(Dependent_count    2.402581
 dtype: float64, Dependent_count    2.335412
 dtype: float64)

Para la cual concluimos que las medias obtenidas a partir de la variable Attrition Flag en cuanto a Dependent Count son diferentes , rechazando así la hipotesis nula en favor de la hipotesis alternativa.

In [ ]:
from scipy import stats

**Para comprobar la vericidad de la diferencia de medias obtenidas anteriormente importamos del paquete scipy a stats. Para poder así realizar el ttest el cual es un test que nos permite hacer una prueba para las medias de dos muestras independientes.**
 
Prueba de hipotesis de diferencia de medias con el ttest:
*   Ho: La diferencia de medias es identica μ1=μ2.
*   Ha: La diferencia de medias es diferente μ1≠μ2.

α=0.05


In [ ]:
stats.ttest_ind(data_attricus, data_exiscus)

Ttest_indResult(statistic=array([1.83195902]), pvalue=array([0.06698689]))

In [ ]:
stats.ttest_ind(data_attridepen, data_exisdepen)

Ttest_indResult(statistic=array([1.91123656]), pvalue=array([0.05600239]))

Para los dos ttest  se puede observar que cada pvalor es mayor a 0,05 por lo tanto,hay evidencias significativas para rechazar la hipotesis nula en favor de las hipotesis alternativas. Por lo tanto las medias de las variables son diferentes

Así mismo tambien realizamos la diferencia de medianas entre las mismas selecciones o divisiones realizadas del data frame.

Prueba de hipotesis de diferencia de medianas:
*   Ho: La diferencia de medianas es igual
*   Ha: La diferencia de medianases diferente

In [ ]:
data_attricus.median(),data_exiscus.median()

(Customer_Age    47.0
 dtype: float64, Customer_Age    46.0
 dtype: float64)

Para esta prueba de hipotesis podriamos concluir quede las medianas obtenidas  a partir de la variable Attrition Flag en cuanto a Customer Age son diferentes, diciendo así que rechazamos la hipotesis nula.

In [ ]:
data_attridepen.median(),data_exisdepen.median()

(Dependent_count    2.0
 dtype: float64, Dependent_count    2.0
 dtype: float64)

Para esta prueba de hipotesis podriamos concluir que las medianas obtenidas  a partir de la variable Attrition Flag en cuanto a Dependent Count son iguales diciendo así que no hay suficiente evidencia estadistica para rechazar la hipotesis nula

#6.Construcción de cinco tablas de contigencia que muestren la relacion por separado de la variable `Attrition Flag` con:
*   Gender
*   Education Level
*   Marital Status
*   Income Category
*   Card Category



En la primera tabla de contigencia, podemos ver la relación entre las variables cualitatitas Attrition Flag y Gender. Encontrando que para las mujeres que se encuentran  en Attrited Customer se tienen 930 datos, para hombres pertenecientes a  Attrited Customer tienen 697 datos, para Existing Customer  Mujer se tiene 4428 datos y para Existing Customer que sea Hombre se tiene 4072 dato. A partir de esto podemos observar que la poblacion de mujeres en general posee una mayor cantidad de datos en comparación con la de los hombres y que la mayor cantidad de clientes existentes son mujeres.

In [ ]:
con_gen=pd.crosstab(data["Attrition_Flag"],data["Gender"])
con_gen

Gender                F     M
Attrition_Flag               
Attrited Customer   930   697
Existing Customer  4428  4072

En la segunda tabla de contigencia, podemos ver la relación entre las variables cualitatitas Attrition Flag y Education Level. Para esta tabla de contingencia observamos que se tienen muchos más datos pertenecientes a Existing Customer, tambien podemos analizar que la relación de mayor magnitud dentro de la tabla de contigencia es entre Existing Customer y Graduate con 2641. Por lo tanto de esta tabla de contigencia podemos afirmar que las personas que tiene un nivel de educación de Graduados representan la mayor población de clientes existentes.

In [ ]:
con_educ=pd.crosstab(data["Attrition_Flag"],data["Education_Level"])
con_educ

Education_Level    College  Doctorate  Graduate  High School  Post-Graduate  \
Attrition_Flag                                                                
Attrited Customer      154         95       487          306             92   
Existing Customer      859        356      2641         1707            424   

Education_Level    Uneducated  Unknown  
Attrition_Flag                          
Attrited Customer         237      256  
Existing Customer        1250     1263

En la tercera tabla de contigencia, podemos ver la relación entre las variables cualitatitas Attrition Flag y Marital Status. Para esta tabla de contingencia  observamos que para Attrited Customer en todos las categorias de Marital Status representan una menor proporció de datos en comparación con Existing Customer. Adicionalmente, la categoría de Married en relación con Existing Customer representan la mayor magnitud de datos con 3275.

In [ ]:
con_marit=pd.crosstab(data["Attrition_Flag"],data["Marital_Status"])
con_marit

Marital_Status     Divorced  Married  Single  Unknown
Attrition_Flag                                       
Attrited Customer       121      709     668      129
Existing Customer       627     3978    3275      620

En la cuarta tabla de contigencia, podemos ver la relación entre las variables cualitatitas Attrition Flag e Income Category. Esta tabla de contigencia tiene la peculiaridad, ya que las categorias dentro de la variable Income son intervalos de valores monetarios. Sin embargo la tabla de contingencia sigue mostrando de forma efectiva la relación entre estas dos variables. Dentro de la relación de estas variables podemos analizar que el intervalo de menos de 40k es el que se relaciona en mayor magnitud con la variable Attrition Flag, tanto para su categoria Attrited Customer con 612 y Existing Customer con 2949. Concluyendo así que las personas que tienen ingresos menores a 40k representn la mayor población de clientes.

In [ ]:
con_incom=pd.crosstab(data["Attrition_Flag"],data["Income_Category"])
con_incom

Income_Category    $120K +  $40K - $60K  $60K - $80K  $80K - $120K  \
Attrition_Flag                                                       
Attrited Customer      126          271          189           242   
Existing Customer      601         1519         1213          1293   

Income_Category    Less than $40K  Unknown  
Attrition_Flag                              
Attrited Customer             612      187  
Existing Customer            2949      925

En la quinta y ultima tabla de contigencia, podemos ver la relación entre las variables cualitatitas Attrition Flag y Card Category. Podemos analizar  que las personas que tiene la categoria de tarjeta Blue representa la mayor población  tanto para Attrited Customer (1519)  como para Existing Customer (7917). 

In [ ]:
con_card=pd.crosstab(data["Attrition_Flag"],data["Card_Category"])
con_card

Card_Category      Blue  Gold  Platinum  Silver
Attrition_Flag                                 
Attrited Customer  1519    21         5      82
Existing Customer  7917    95        15     473

#6.Lleve a cabo, para cada tabla, una prueba $\chi^2$

**PRUEBA DE HIPOTESIS PARA TODAS LAS** $\chi^2$
* Hipotesis Nula: No Existe una relación significativa entre las dos variables.
* Hipotesis Alterna: Existe una relación significativa entre las dos variables.

α=0.05

**Para la tabla de contingencia entre Attition Flag y Gender**

In [ ]:
st.chi2_contingency(con_gen) 

(13.865613700637342,
 0.00019635846717310269,
 1,
 array([[ 860.81425891,  766.18574109],
        [4497.18574109, 4002.81425891]]))

Se rechaza la hipotesis nula y se comprueba que si existe relación significativa entre las variables Attrtition Flag y Gender.

**Para la tabla de contingencia entre Attition Flag y Education Level**

In [ ]:
st.chi2_contingency(con_educ) 

(12.51120540483051,
 0.05148913147336627,
 6,
 array([[ 162.74819789,   72.45748988,  502.54330009,  323.40782068,
           82.90036536,  238.90085909,  244.04196702],
        [ 850.25180211,  378.54251012, 2625.45669991, 1689.59217932,
          433.09963464, 1248.09914091, 1274.95803298]]))

No existe suficiente evidencia estadistica para aceptar la hipotesis alterna, mostrando así que no existe relación significativa entre las variables Attrition Flag y Education Level.

**Para la tabla de contingencia entre Attition Flag y Marital Status**

In [ ]:
st.chi2_contingency(con_marit) 

(6.056097396726619,
 0.10891263394840227,
 3,
 array([[ 120.17339785,  753.01165202,  633.48089266,  120.33405747],
        [ 627.82660215, 3933.98834798, 3309.51910734,  628.66594253]]))

No existe suficiente evidencia estadistica para aceptar la hipotesis alterna, nos quedamos con la hipotesis nula de que no existe relación significativa entre las variables Attrition Flag y Marital Status.

**Para la tabla de contingencia entre Attition Flag e Income Category**

In [ ]:
st.chi2_contingency(con_incom) 

(12.832259274756895,
 0.025002425704390617,
 5,
 array([[ 116.79954577,  287.5807248 ,  225.24479115,  246.61252098,
          572.10891676,  178.65350054],
        [ 610.20045423, 1502.4192752 , 1176.75520885, 1288.38747902,
         2988.89108324,  933.34649946]]))

Se rechaza la hipotesis nula a favor de la hipotesis alterna, conluyendo así que existe una relación significativa entre las variables Attriiton Flag y Income Category.

**Para la tabla de contingencia entre Attition Flag y Card Category**

In [ ]:
st.chi2_contingency(con_card) 

(2.2342267000046805,
 0.5252382797994759,
 3,
 array([[1.51598420e+03, 1.86365162e+01, 3.21319246e+00, 8.91660906e+01],
        [7.92001580e+03, 9.73634838e+01, 1.67868075e+01, 4.65833909e+02]]))

No existe suficiente evidencia estadistica para aceptar la hipotesis alterna, nos quedamos con la hipotesis nula que nos permite concluir que existe una relación significativa entre Attrition Flag y Card Category.

#Conclusión General

A partir del analisis del dataset en probabilidad y estadistica inferencial podemos concluir que las variables  ``` Attrition Flag ```  y  ``` Gender```presentan unas proporciones significativas . Así mismo verificamos y comprobamos a traves del estadistico de prueba de shapiro que ni la variable Customer Age ni Dependent Count tiene una distribucción normal. Tambien se realizo una división del dataset que permitió verificar la diferencia de medias en cuanto a las variables Customer Age y Dependent Count, los resultados se encuentran dentro del contenido. Por ultimo se realizo una prueba chi 2 en las tablas de contingencia previamente creadas , lo cual nos permitio conlcuir sobre la relación significativa entre las variables dadas.  

